bin/deploy

```diff
@@ -169,23 +176,47 @@ main() {
   local conda_packages
   conda_packages="conda${pyver_prefix}_${deploy_mode}-${pkg_postfix}.yml"
 
+  if [[ $FIXED_ENVREF != '' ]]; then
+    # (try to) get latest branch SHA1 and assign it to LSST_SPLENV_REF
+    # if the REF is not a branch (or a tag), the reftip will be empty
+    reftip=`git ls-remote https://github.com/lsst/scipipe_conda_env.git ${ENVREF} | cut -c1-7`
+    if [[ $reftip == '' ]]; then
+        # the provided ref is not a branch nor a tag
+        LSST_SPLENV_REF="${ENVREF}"
+        # Defining environment name based on the provided SHA1
+        LSST_CONDA_ENV_NAME="scipipe-env-${FIXED_ENVREF}"
+        local env_file="${LSSTSW}/env/${ENVREF}/${conda_packages}"
+    else
+        LSST_SPLENV_REF="$reftip"
+        # Defining environment name based on branch and SHA1 from the tip of the branch
+        LSST_CONDA_ENV_NAME="scipipe-env-${FIXED_ENVREF}.${LSST_SPLENV_REF}"
```

Comment:
In this way, you will know more precisely which environment you are using.

You mean, only to store one environment file and overwrite it?
In this way, we have a local copy of the environment file ready to use from the command line, for debug or manual comparison with other environments.

Reply:
No, I don't mean to only have one environment.  The `LSST_SPLENV_REF` is already unique, though, so I wasn't sure why you needed to add `FIXED_ENVREF` to it.

Reply:
`FIXED_ENVREF` is a mangled version of the supplied `-r` argument?  The logic is therefore that if `-r` is a tag name then you'll use that tag name directly, if it's a branch name you are using the name of the branch and the explicit sha1 associated with that branch? This is so that when someone reports what env they are using they indicate that it came from a branch request? Seems okay to me